src/import/sst_service.rs

```diff
@@ -966,51 +1011,41 @@ fn build_apply_request<'a, 'b>(
 where
     'a: 'b,
 {
-    let mut req_size = 0_u64;
-
     // use callback to collect kv data.
-    if is_delete {
-        Box::new(move |k: Vec<u8>, _v: Vec<u8>| {
-            let mut req = Request::default();
-            let mut del = DeleteRequest::default();
+    Box::new(move |k: Vec<u8>, v: Vec<u8>| {
+        let mut req = Request::default();
 
+        if is_delete {
+            let mut del = DeleteRequest::default();
             del.set_key(k);
             del.set_cf(cf.to_string());
             req.set_cmd_type(CmdType::Delete);
             req.set_delete(del);
-            req_size += req.compute_size() as u64;
-            reqs.accept(req);
-            // When the request size get grow to half of the max request size,
-            // build the request and add it to a batch.
-            if req_size > raft_size / 2 {
-                req_size = 0;
-                let cmd = make_request(reqs, context.clone());
-                cmd_reqs.push(cmd);
-            }
-        })
-    } else {
-        Box::new(move |k: Vec<u8>, v: Vec<u8>| {
+        } else {
             if cf == CF_WRITE && !write_needs_restore(&v) {
                 return;
             }
 
-            let mut req = Request::default();
             let mut put = PutRequest::default();
-
             put.set_key(k);
             put.set_value(v);
             put.set_cf(cf.to_string());
             req.set_cmd_type(CmdType::Put);
             req.set_put(put);
-            req_size += req.compute_size() as u64;
-            reqs.accept(req);
-            if req_size > raft_size / 2 {
-                req_size = 0;
-                let cmd = make_request(reqs, context.clone());
-                cmd_reqs.push(cmd);
-            }
-        })
-    }
+        }
+
+        // When the request size get grow to max request size,
+        // build the request and add it to a batch.
+        if *req_size + req.compute_size() as u64 > raft_size * 7 / 8 {
```

Comment:
why change the raft_size here? will it cause error `if req.compute_size() > 1/8 * raft_size`?